### PR TITLE
Add System.IO.FileSystem and System.IO.FileSystem.Primitives

### PR DESCRIPTION
**Pull request by ericstj**
This change adds the FileSystem and FileSystem.Primitives assemblies.  I have written new tests for FS.Primitives and created a starting point for porting existing tests into FileSystem.  I'd like to hold off on any contributions to FileSystem until I flesh out the test suite.

### CODE REVIEW

**Comment by ericstj** [comment]
Fixed the build break, it happened as a result of fat-fingering a per-line commit.  All should be good now.

**Comment by terrajobst** [comment]
LGTM

File: src/System.IO.FileSystem/src/System/IO/ReadLinesIterator.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                // To maintain 4.0 behavior we Dispose 
+                // after reading to the end of the reader.
+                Dispose();
+            }
+
+            return false;
+        }
+
+        protected override Iterator<string> Clone()
+        {
```
Comment (FiveTimesTheFun) [inline src/System.IO.FileSystem/src/System/IO/ReadLinesIterator.cs]: This comment is stale (see DevDiv 11141)

**Comment by FiveTimesTheFun** [comment]
Looks great thanks Eric :)

**Pull request merged**